lib/redmine_required_field/patches/issues_controller_patch.rb

```diff
@@ -36,6 +36,30 @@ def update
 
 
       def rrf_params_match_settings(rrf_setting)
+        #check role and assignee behavior(negate match for given roles if checkbox is set, and only apply filter if attached to only is set and user matches attached to )
+        user_is_assigned_to = User.current.id==@issue.assigned_to_id
+        if rrf_setting[RedmineRequiredField::ROLE_EXCLUDED] == '1'
+
+          if rrf_setting[RedmineRequiredField::CHECK_ASSIGNEE_ONLY] =='1'
+            if !user_is_assigned_to
+              return false
+            else
+              setting_role_ids = rrf_setting[RedmineRequiredField::ROLE_IDS]
+              matched = !RedmineRequiredField.setting_values_contain_ids(setting_role_ids, RedmineRequiredField.current_roles)
+            return false unless matched
+            end
+          end
+        else
+          if rrf_setting[RedmineRequiredField::CHECK_ASSIGNEE_ONLY]='1'
+            if !user_is_assigned_to
+              return false
+            else
+              setting_role_ids = rrf_setting[RedmineRequiredField::ROLE_IDS]
+              matched = RedmineRequiredField.setting_values_contain_ids(setting_role_ids, RedmineRequiredField.current_roles)
+            return false unless matched
+            end
+          end
+        end
```

Comment:
The whole block above has a lot of cyclometric complexity. It can be simplified by using guard clauses. E.g. converting this

    if aaa && bbb
      if ccc
         return false
      else
        foo = bar
      end
    else

to this

    if aaa && bbb
      return false if ccc

      foo = bar
    else

Another thing is the if negative, so since we are using "else" writing this

    if a
      foo
    else
       bar
    end

is better than writing this

    if !a
      bar
    else
       foo
    end

One more thing, since you are running very similar code, with different variable values (setting_role_ids for example) how about setting up the variables at first, and then run the setting_values_contain_ids at the end ? That should simplify the code too.

Reply:
i used the guard clauses to simplify the code.
simplified the logical cycles